projects/plugins/jetpack/extensions/blocks/subscriber-login/subscriber-login.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * Subscriber Login Block.
+ *
+ * @since 0.0.1
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Extensions\Subscriber_Login;
+
+use Automattic\Jetpack\Blocks;
+use Automattic\Jetpack\Extensions\Premium_Content\Subscription_Service\Abstract_Token_Subscription_Service;
+use Jetpack_Gutenberg;
+use Jetpack_Memberships;
+
+/**
+ * Registers the block for use in Gutenberg
+ * This is done via an action so that we can disable
+ * registration if we need to.
+ */
+function register_block() {
+	if ( ! \Jetpack::is_module_active( 'subscriptions' ) || ! class_exists( '\Jetpack_Memberships' ) ) {
+		return;
+	}
+
+	Blocks::jetpack_register_block(
+		__DIR__,
+		array( 'render_callback' => __NAMESPACE__ . '\render_block' )
+	);
+
+	add_action( 'wp_logout', __NAMESPACE__ . '\subscriber_logout' );
+}
+add_action( 'init', __NAMESPACE__ . '\register_block' );
+
+/**
+ * Logout subscriber.
+ *
+ * @return void
+ */
+function subscriber_logout() {
+	Abstract_Token_Subscription_Service::clear_token_cookie();
+}
+
+/**
+ * Returns current URL.
+ *
+ * @return string
+ */
+function get_current_url() {
+	if ( ! isset( $_SERVER['HTTP_HOST'] ) || ! isset( $_SERVER['REQUEST_URI'] ) ) {
+		return '';
+	}
+
+	return ( is_ssl() ? 'https://' : 'http://' ) . wp_unslash( $_SERVER['HTTP_HOST'] ) . wp_unslash( $_SERVER['REQUEST_URI'] ); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+}
+
+/**
+ * Renders Subscriber Login block.
+ *
+ * @param array $attr    Array containing the Subscriber Login block attributes.
+ *
+ * @return string
+ */
+function render_block( $attr ) {
+	Jetpack_Gutenberg::load_assets_as_required( __DIR__ );
+
+	if ( ! is_user_logged_in() ) {
```

Comment:
Is this enough to check if the user is logged in or not? Asking because I noticed this:
https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php#L557-L561

Reply:
I think you'd need something like this:

https://github.com/Automattic/jetpack/blob/20e0d8fe09b53c71910a9e31e68b28d7618ac6b0/projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php#L555-L561

Reply:
Oh yeah, this could also work, and I think better:

https://github.com/Automattic/jetpack/blob/0f70c4ed3ac8d90761ef59d34a726d6419a5604c/projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php#L1096-L1114